var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -85,31 +93,42 @@ def get_rocm_prefix_info(self):
                 raise RuntimeError(msg)
 
             return {
+                'rocm-path': fallback_prefix,
                 'llvm-amdgpu': fallback_prefix.llvm,
                 'hsa-rocr-dev': fallback_prefix.hsa,
                 'rocminfo': fallback_prefix.bin,
                 'rocm-device-libs': fallback_prefix,
             }
         else:
-            return dict((name, self.spec[name].prefix)
-                        for name in ('llvm-amdgpu', 'hsa-rocr-dev', 'rocminfo',
-                                     'rocm-device-libs'))
+            mydict = dict((name, self.spec[name].prefix)
+                          for name in ('llvm-amdgpu', 'hsa-rocr-dev',
+                                       'rocminfo', 'rocm-device-libs'))
+            mydict['rocm-path'] = os.path.dirname(self.spec.prefix)
+            return mydict
 
     def setup_dependent_build_environment(self, env, dependent_spec):
         # Indirection for dependency paths because hip may be an external in
-        # Spack. See block comment on get_rocm_prefix_info
+        # Spack. See block comment on get_rocm_prefix_info .
+
+        # NOTE: DO NOT PUT LOGIC LIKE self.spec[name] in this function!!!!!
+        # It DOES NOT WORK FOR EXTERNAL PACKAGES!!!! See get_rocm_prefix_info
         rocm_prefixes = self.get_rocm_prefix_info()
 
-        env.set('ROCM_PATH', '')
+        env.set('ROCM_PATH', rocm_prefixes['rocm-path'])
         env.set('HIP_COMPILER', 'clang')
         env.set('HIP_PLATFORM', 'hcc')
         env.set('HIP_CLANG_PATH', rocm_prefixes['llvm-amdgpu'].bin)
         env.set('HSA_PATH', rocm_prefixes['hsa-rocr-dev'])
         env.set('ROCMINFO_PATH', rocm_prefixes['rocminfo'])
-        env.set('DEVICE_LIB_PATH',
-                self.spec['rocm-device-libs'].prefix.amdgcn.bitcode)
+        env.set('DEVICE_LIB_PATH', rocm_prefixes['rocm-device-libs'].lib)
+        env.set('HIP_PATH', rocm_prefixes['rocm-path'])
```

Comment:
This is probably incorrect @dtaller, can you please test with all packages installed by spack instead of by the system?